src/service-config.ts

```diff
@@ -0,0 +1,67 @@
+import ProgramArguments from "./program-arguments";
+
+export enum EndpointType {
+    CLIENT = "client",
+    SERVER = "server"
+};
+
+export default class ServiceConfiguration {
+    private _endpointType: EndpointType = EndpointType.SERVER;
+    private _forceMockI2C: boolean = false;
+    private _port: number = 8080;
+    private _host: string = "localhost";
+    private _uri: string = "/wpilibws";
+
+    constructor(programArgs: ProgramArguments) {
+        if (programArgs.endpointType !== "client" && programArgs.endpointType !== "server") {
```

Comment:
Technically speaking we should probably add a if(!programArgs) to error when programArgs is undefined.